packages/server/lib/browsers/protocol.js

```diff
@@ -0,0 +1,75 @@
+const _ = require('lodash')
+const CRI = require('chrome-remote-interface')
+const { connect } = require('@packages/network')
+const Promise = require('bluebird')
+const la = require('lazy-ass')
+const is = require('check-more-types')
+const pluralize = require('pluralize')
+const debug = require('debug')('cypress:server:protocol')
+
+function getDelayMsForRetry (i) {
```

Comment:
i believe this function will cause the retrying socket to never stop retrying lol

where mah tests? the tests should indicate it continues to retry because `getDelayMsForRetry` short circuits the retry loop when it returns `undefined`. returning a value will always cause it to retry.

Reply:
It stops returning a value after the 10th iteration, which should end the retry cycle.

Reply:
https://github.com/cypress-io/cypress/blob/70bdc7d89984bff08487c280b6bc23d92458c134/packages/network/lib/connect.ts#L82-L88

Reply:
yeah that's true - i misread the code as: "always return a value". still missing a test though - you can borrow from the existing network tests that test `getDelayMsForRetry` that do precisely this.

Reply:
ddb168d2ad48ebbec546c01a3feea77351fab745